card/lib/card/set/event.rb

```diff
@@ -1,21 +1,27 @@
 class Card
-  def deserialize_for_active_job! attr, env, current_id
+  def deserialize_for_active_job! attr
     attr.each do |attname, args|
       # symbols are not allowed so all symbols arrive here as strings
       # convert strings that were symbols before back to symbols
       value = args[:symbol] ? args[:value].to_sym : args[:value]
       instance_variable_set("@#{attname}", value)
     end
     include_set_modules
+  end
+
+  def with_env_and_auth env, auth
     # If active jobs (and hence the integrate_with_delay events) don't run
     # in a background process then Card::Env.deserialize! decouples the
     # controller's params hash and the Card::Env's params hash with the
     # effect that params changes in the CardController get lost
     # (a crucial example are success params that are processed in
     # CardController#update_params_for_success)
-    return if Wagn.config.active_job.queue_adapter == :inline
-    Card::Env.deserialize! env
-    Card::Auth.current_id = current_id
+    return yield if Wagn.config.active_job.queue_adapter == :inline
+    Card::Auth.deserialize auth do
```

Comment:
I like the name of the method defined here and am tempted to suggest just `Card::Auth.with(auth)`